monitor/uidmonitor/processor.go

```diff
@@ -0,0 +1,398 @@
+type StoredContext struct {
+	MarkVal   string
+	EventInfo *rpcmonitor.EventInfo
+}
+
+// NewCustomUIDProcessor initializes a processor with a custom path
+func NewCustomUIDProcessor(storePath string, collector collector.EventCollector, puHandler monitor.ProcessingUnitsHandler, metadataExtractor rpcmonitor.RPCMetadataExtractor, releasePath string) *UIDProcessor {
+
+	return &UIDProcessor{
+		collector:         collector,
+		puHandler:         puHandler,
+		metadataExtractor: metadataExtractor,
+		netcls:            cgnetcls.NewCgroupNetController(releasePath),
+		contextStore:      contextstore.NewContextStore(storePath),
+		storePath:         storePath,
+		regStart:          regexp.MustCompile("^[a-zA-Z0-9_].{0,11}$"),
+		regStop:           regexp.MustCompile("^/trireme/[a-zA-Z0-9_].{0,11}$"),
+		puToPidEntry:      cache.NewCache(),
+		pidToPU:           cache.NewCache(),
+	}
+}
+
+// NewUIDProcessor creates a default Linux processor with the standard trireme path
+func NewUIDProcessor(collector collector.EventCollector, puHandler monitor.ProcessingUnitsHandler, metadataExtractor rpcmonitor.RPCMetadataExtractor, releasePath string) *UIDProcessor {
+	return NewCustomUIDProcessor("/var/run/trireme/linux", collector, puHandler, metadataExtractor, releasePath)
+}
+
+// Create handles create events
+func (s *UIDProcessor) Create(eventInfo *rpcmonitor.EventInfo) error {
+
+	return s.puHandler.HandlePUEvent(eventInfo.PUID, monitor.EventCreate)
+}
+
+// Start handles start events
+func (s *UIDProcessor) Start(eventInfo *rpcmonitor.EventInfo) error {
+	s.Lock()
+	defer s.Unlock()
+	contextID := eventInfo.PUID
+	pids, err := s.puToPidEntry.Get(contextID)
+	var runtimeInfo *policy.PURuntime
+	if err != nil {
+		runtimeInfo, err = s.metadataExtractor(eventInfo)
+		if err != nil {
+			return err
+		}
+		publishedContextID := contextID + runtimeInfo.Options().CgroupMark
+		// Setup the run time
+		if err = s.puHandler.SetPURuntime(publishedContextID, runtimeInfo); err != nil {
+			return err
+		}
+
+		defaultIP, _ := runtimeInfo.DefaultIPAddress()
+
+		zap.L().Error("Starting ", zap.String("contextID", contextID), zap.String("Publishing", publishedContextID))
+		if perr := s.puHandler.HandlePUEvent(publishedContextID, monitor.EventStart); perr != nil {
+			zap.L().Error("Failed to activate process", zap.Error(perr))
+			return perr
+		}
+
+		err = s.processLinuxServiceStart(eventInfo, runtimeInfo)
+
+		if err != nil {
+			zap.L().Error("ProcessLInuxServiceStart", zap.Error(err))
+			return err
+		}
+
+		s.collector.CollectContainerEvent(&collector.ContainerRecord{
+			ContextID: contextID,
+			IPAddress: defaultIP,
+			Tags:      runtimeInfo.Tags(),
+			Event:     collector.ContainerStart,
+		})
+		entry := &putoPidEntry{
+			Info:               runtimeInfo,
+			publishedContextID: publishedContextID,
+		}
+		entry.pidlist = make(map[string]bool, 20)
+		entry.pidlist[eventInfo.PID] = true
```

Comment:
it’s more efficient to do a map[string]struct{} than a map[string]bool if you want to emulate a set